### PR TITLE
fix: pass http.Agent through `connect`

### PR DESCRIPTION
**Pull request by khoerling**
Allow web sockets to use agents, also.

### CODE REVIEW

**Comment by khoerling** [comment]
Let me know any changes, fairly small & simple request-- cheers.

File: src/websocket-client.ts [hunk 1]
```diff
@@ -510,7 +510,8 @@ export class WebsocketClient extends EventEmitter {
   private connectToWsUrl(url: string, wsKey: WsKey): WebSocket {
     this.logger.silly(`Opening WS connection to URL: ${url}`, { ...loggerCategory, wsKey })
 
-    const ws = new WebSocket(url);
+    const agent = this.options.requestOptions?.agent;
+    const ws = new WebSocket(url, undefined, { agent });
```
Comment (tiagosiebler) [inline src/websocket-client.ts]: Looks good - could you check if there's any change in behaviour if "agent" happens to be undefined from that optional chain? Does it still pass the same useragent as before? websockets still work fine, private & public, if requestOptions is undefined?

Comment (khoerling) [reply src/websocket-client.ts]: It did work from our local testing.  I went ahead with rebasing against the latest and added a trinary to keep that original value as-is.  Let me know,

**Review by tiagosiebler — APPROVED** [review]
Looks good, thanks!

File: src/websocket-client.ts (hunk 1 above)
Comment (tiagosiebler) [reply src/websocket-client.ts]: Thanks - just wanted to make sure. It'll be available in a few minutes on the latest version, v2.1.5

**Pull request merged**